src/snowflake/sqlalchemy/snowdialect.py

```diff
@@ -226,10 +226,33 @@ def create_connect_args(self, url):
             opts["host"] = opts["host"] + ".snowflakecomputing.com"
             opts["port"] = "443"
         opts["autocommit"] = False  # autocommit is disabled by default
-        opts.update(url.query)
+
+        query = dict(**url.query)  # make mutable
```

Comment:
Why do we need a mutable instance?

Reply:
In order to `.pop` out the parameters that are meant for `snowflake-sqlalchemy` itself, rather than the driver.